p2p/net/upgrader/upgrader.go

```diff
@@ -201,9 +201,7 @@ func (u *upgrader) upgrade(ctx context.Context, t transport.Transport, maconn ma
 
 func (u *upgrader) setupSecurity(ctx context.Context, conn net.Conn, p peer.ID, dir network.Direction) (sec.SecureConn, protocol.ID, bool, error) {
```

Comment:
You don't need the third return, bool value right? 

AFAICT it is used to indicate whether we are the server which the caller now has complete context of.

Reply:
Good catch. This allowed a small refactor to simplify things a tiny bit.